astropy/units/format/ogip.py

```diff
@@ -344,16 +344,21 @@ def _format_superscript(cls, number: str) -> str:
 
     @classmethod
     def to_string(
-        cls, unit: UnitBase, fraction: bool | Literal["inline", "multiline"] = "inline"
+        cls,
+        unit: CompositeUnit | NamedUnit,
+        fraction: bool | Literal["inline", "multiline"] = "inline",
     ) -> str:
         # Remove units that aren't known to the format
-        unit = cls._decompose_to_known_units(unit)
+        base_unit = cls._decompose_to_known_units(unit)
 
         if isinstance(unit, CompositeUnit):
             # Can't use np.log10 here, because p[0] may be a Python long.
-            if math.log10(unit.scale) % 1.0 != 0.0:
+            if (
+                not isinstance(base_unit.scale, float)
```

Comment:
Have you checked if the OGIP format even allows complex scales?

Reply:
I think for this PR perhaps just ensuring that a warning is raised if the scale is not float would be appropriate. It is a good question, though, whether it shouldn't be an error...

Reply:
I'm taking the existing warning at face value here: if the scale needs to be a power of 10, it cannot be complex.